src/pudl/transform/ferc1.py

```diff
@@ -5851,21 +5901,109 @@ def add_dimension_total_calculations(
                 weight=1,
             )
         )
-
         subtotal_dfs.append(
             total_w_subdim.assign(**{f"is_within_dimension_{dim}": True})
         )
     # now we have a bunch of *new* records linking total records to their non-total
     # sub-components.
-    calc_components = pd.concat(
-        [
+    calc_components = pd.concat([calc_components, pd.concat(subtotal_dfs)]).reset_index(
```

Comment:
Why the nested concats? Would this work just as well?
```py
    calc_components = pd.concat(subtotal_dfs + [calc_components]).reset_index(drop=True)
```

Reply:
oh good point this was not necessary